client/my-sites/email/email-providers-in-depth-comparison/comparison-table/index.tsx

```diff
@@ -1,21 +1,211 @@
+import { Gridicon } from '@automattic/components';
+import classnames from 'classnames';
+import { translate } from 'i18n-calypso';
 import { FunctionComponent } from 'react';
+import InfoPopover from 'calypso/components/info-popover';
+import Main from 'calypso/components/main';
+import { FullWidthButton } from 'calypso/my-sites/marketplace/components';
 import type { TranslateResult } from 'i18n-calypso';
 
+import './style.scss';
+
 export type EmailProviderFeatures = {
+	id: string;
 	header: string;
+	headerLogo: string;
+	headerLogoIsGridIcon?: boolean;
 	tools: TranslateResult;
 	storage: TranslateResult;
 	importing: TranslateResult;
+	selectCallback?: () => void;
+	subtitle: TranslateResult;
 	support: TranslateResult;
-	selectCallback: () => void;
+	footerBadge?: string;
+	learnMore: string | null;
 };
 
 type ComparisonTableProps = {
+	className: string;
 	emailProviders: EmailProviderFeatures[];
+	showFeatureNames?: boolean;
 };
 
-const ComparisonTable: FunctionComponent< ComparisonTableProps > = () => {
-	return <p> New component </p>;
+const ComparisonTable: FunctionComponent< ComparisonTableProps > = ( {
+	className,
+	emailProviders,
+	showFeatureNames = true,
+} ) => {
+	const showCallbackRow = emailProviders.some( ( provider ) => provider.selectCallback );
+	const logoComponentProvider = ( provider: EmailProviderFeatures ) =>
+		provider.headerLogoIsGridIcon ? (
+			<Gridicon className="comparison-table__header-logo" icon={ provider.headerLogo } />
+		) : (
+			<img
+				alt={ provider.header }
+				src={ provider.headerLogo }
+				className="comparison-table__header-logo"
+			/>
+		);
```

Comment:
That seems convoluted to me, what if we simply pass that code when the list of features is defined?

```javascript
export const professionalEmailFeatures: EmailProviderFeatures = {
  logo: <Gridicon icon="my-sites" />,
};

export const googleWorkspaceFeatures: EmailProviderFeatures = {
  logo: <img src={ googleWorkspaceIcon } />,
};
```